modules/cicd/argo/modules/cd/main.tf

```diff
@@ -1,30 +1,132 @@
-resource "helm_release" "argo-cd" {
-  depends_on = [
-    var.module_depends_on
-  ]
+resource kubernetes_namespace this {
+  count = var.namespace == "" ? 1 : 0
+  metadata {
+    name = var.namespace_name
+  }
+}
 
-  name          = "argocd"
-  repository    = "https://argoproj.github.io/argo-helm"
-  chart         = "argo-cd"
-  version       = "2.3.5"
-  namespace     = var.namespace
+resource helm_release this {
+  name          = local.name
+  repository    = local.repository
+  chart         = local.chart
+  version       = var.chart_version
+  namespace     = local.namespace
   recreate_pods = true
   timeout       = 1200
-  values = [
-    file("${path.module}/values.yml")
-  ]
-  dynamic set {
-    for_each = merge(local.cd_conf_defaults, var.cd_conf)
 
+  dynamic set {
+    for_each = merge(var.enabled ? merge(local.init_conf, local.conf) : local.legacy_defaults, var.cd_conf)
     content {
       name  = set.key
       value = set.value
     }
   }
 }
 
+data aws_region current {}
+
+data aws_eks_cluster this {
+  name = var.cluster_name
```

Comment:
if cluster not exist - terraform plan has error ResourceNotFoundException: No cluster found for name